### PR TITLE
adds rack config for sidekiq-web

### LINKED ISSUE #323: Add Sidekiq UI for viewing progress
Provide a UI for viewing Sidekiq workers.

## What does done look like

- Sidekiq Admin interface at `/sidekiq` from the catalog. This is webauth protected, only ul-dev-team can display it.

### PR DESCRIPTION
**Pull request by whereismyjetpack**
in k8s, we can run sidekiq-web at `/sidekiq` configure rack here. 

Fixes #323

### CODE REVIEW

File: lib/psulib_traject/workers/indexer.rb [hunk 1]
```diff
@@ -6,10 +6,10 @@ class Indexer < Base
       # @param [Hash] args
       # @option args [String, Pathname] :filename (required)
       # @option args [String] :collection_name
-      def perform(filename, opts = {})
+      def perform(filename, collection_name = nil)
```
Comment (awead) [inline lib/psulib_traject/workers/indexer.rb:9]: was there a reason the hash parameter wouldn't work here? It's better to have optional parameters as a hash so you don't have to worry about positionality later on.

File: config.ru [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'sidekiq'
+require 'sidekiq-scheduler/web'
```
Comment (awead) [inline config.ru:2]: sidekiq and sidekiq-scheduler are being required in psublib_traject. are these needed here?

File: config.ru [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+Sidekiq.configure_client do |config|
+  config.redis = { :size => 1 }
+end
+
+require 'sidekiq/web'
```
Comment (awead) [inline config.ru:8]: move this up with the other requires, or put them in publib_traject.rb.

**Review by awead — CHANGES_REQUESTED** [review]
I have some questions about the require steps. We might need to move some around between this config.ru and psulib_traject.rb

File: lib/psulib_traject/workers/indexer.rb (hunk 1 above)
Comment (whereismyjetpack) [reply lib/psulib_traject/workers/indexer.rb:9]: no idea, when called from rake the argument gets ignored. 

`bundle exec rake traject:index['foo','bar']`  bar gets ignored.

File: config.ru (hunk 2 above)
Comment (whereismyjetpack) [reply config.ru:2]: only the /web requires are in config now

**Comment by whereismyjetpack** [comment]
@banukutlu I think this is ready now. I've re-arranged the requires, (imports?) and tested it on my preview branch, https://catalog-djb44.dev.k8s.libraries.psu.edu/sidekiq

**Comment by banukutlu** [comment]
> @banukutlu I think this is ready now. I've re-arranged the requires, (imports?) and tested it on my preview branch, https://catalog-djb44.dev.k8s.libraries.psu.edu/sidekiq

kewl thanks, I'm on it 👍🏻

**Review by banukutlu — APPROVED** [review]
lgtm 👍🏻

File: lib/psulib_traject/workers/indexer.rb (hunk 1 above)
Comment (awead) [reply lib/psulib_traject/workers/indexer.rb:9]: I'd still like to have an opts hash here. Which test was failing?

Comment (awead) [reply lib/psulib_traject/workers/indexer.rb:9]: OK, this is fixed now, but the rake tasks needed to be updated to reflect the different arguments.

**Review by awead — APPROVED** [review]

File: lib/psulib_traject/workers/indexer.rb (hunk 1 above)
Comment (whereismyjetpack) [reply lib/psulib_traject/workers/indexer.rb:9]: I misspoke, _shocker_ i know. The issue was rake sends the argument as a string, and rspec sends the argument as a symbol, so when the tests were passing the method was ignoring the argument that the rake test was passing, i guess the rake task could do something like args.collection.to_sym? or something. feels hacky

**Comment by awead** [comment]
@banukutlu My change to the rake tasks removed your previous approval, so I just went ahead, re-approved, and merged.

File: lib/psulib_traject/workers/indexer.rb (hunk 1 above)
Comment (awead) [reply lib/psulib_traject/workers/indexer.rb:9]: Yeah, I just updated the rake tasks with the new positional arguments.

**Pull request merged**